app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -360,4 +343,33 @@ private int getMemoryIncreaseForGC() {
       return 0;
     }
   }
+
+  private void suggestExecutorInstances() {
+    if (suggestedCore != lastRunExecutorCore) {
+      suggestExecutorInstancesBasedOnCore();
+    } else {
+      suggestedExecutorInstances = lastRunExecutorInstances;
+    }
+    if (suggestedExecutorInstances > CLUSTER_DEFAULT_DYNAMIC_ALLOCATION_MAX_EXECUTOR) {
+      suggestedExecutorInstances = CLUSTER_DEFAULT_DYNAMIC_ALLOCATION_MAX_EXECUTOR;
+    }
+    //job fails if max executors is less than executor instances
+    if (lastRunDynamicAllocationMaxExecutors != null
+        && suggestedExecutorInstances > lastRunDynamicAllocationMaxExecutors) {
+      suggestedExecutorInstances = lastRunDynamicAllocationMaxExecutors;
+    }
+  }
+
+  private void suggestExecutorInstancesBasedOnCore() {
+    Integer lastRunActualExecutorInstances = lastRunExecutorInstances;
+    if (lastRunDynamicAllocationEnabled && lastRunDynamicAllocationMaxExecutors != null) {
+      if (lastRunActualExecutorInstances == null) {
+        lastRunActualExecutorInstances = lastRunDynamicAllocationMaxExecutors;
```

Comment:
Is it fine to consider max executors as last run executors? lastRunDynamicAllocationMaxExecutors is entirely based on config value if I am not wrong. Unless this value is modified based on Utils.scala code to get max executor instances, this would remain 900.
Probably do not suggest executor instances if we do that.

Reply:
Updated the logic and removed use of lastRunDynamicAllocationMaxExecutors.